core/src/replay_stage.rs

```diff
@@ -3046,6 +3047,51 @@ impl ReplayStage {
                     }
                 }
 
+                if bank.collector_id() != my_pubkey {
+                    // If the block does not have at least DATA_SHREDS_PER_FEC_BLOCK shreds in the last FEC set,
+                    // mark it dead. No reason to perform this check on our leader block.
+                    if !blockstore
+                        .is_last_fec_set_full(bank.slot())
+                        .inspect_err(|e| {
+                            warn!(
+                                "Unable to determine if last fec set is full for slot {} {},
+                                marking as dead: {e:?}",
+                                bank.slot(),
+                                bank.hash()
+                            )
+                        })
+                        .unwrap_or(false)
+                    {
+                        // Update metric regardless of feature flag
+                        datapoint_warn!(
+                            "incomplete_final_fec_set",
+                            ("slot", bank_slot, i64),
+                            ("hash", bank.hash().to_string(), String)
```

Comment:
I am told submitting these `hash` values are expensive for metrics database because they don't compress.
But I am guessing these will be very infrequent, so should be fine.